### ISSUE TITLE
clang: error: no such file or directory: '/Users/user/Library/Developer/Xcode/DerivedData/client-xxx/Build/Products/Debug-iphonesimulator/libRNIap.a' Command Ld failed with a nonzero exit code

### ISSUE THREAD

**Issue by booboothefool** [dooboolab/react-native-iap#1174]
I haven't run my project's code since April (though it's live on the app store and works) and am trying to get it in a development state again. When building I am running into:

```
clang: error: no such file or directory: '/Users/user/Library/Developer/Xcode/DerivedData/client-xxx/Build/Products/Debug-iphonesimulator/libRNIap.a' Command Ld failed with a nonzero exit code
```

```
    "react": "16.8.3",
    "react-native": "https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz",
    "react-native-iap": "2.3.26",
```

Any advice would be appreciated.

**Comment by hyochan** [comment]
Looks like you are using expo managed workflow. It won't work in that environment.

**Comment by booboothefool** [comment]
@hyochan Hi I am actually using detached Expo. So I've successfully gotten RNIap to work at one point. For example, today there have been in-app purchases from my users. But it's been a few months and I'm trying to figure out what changed in order to get my project to build again.

**Comment by booboothefool** [comment]
@hyochan Please help me. I have been struggling to get this to run again. The rest of my project appears to build ok except RNIap.

When I add RNIap to my libraries, it turns red:
![image](https://user-images.githubusercontent.com/13257645/98453135-da998c00-2123-11eb-86c4-fe6f6640b85e.png)
![image](https://user-images.githubusercontent.com/13257645/98453145-f2711000-2123-11eb-884e-abf119aee92f.png)
![image](https://user-images.githubusercontent.com/13257645/98453184-4d0a6c00-2124-11eb-98f8-b20558fd93d5.png)

**Comment by hyochan** [comment]
@booboothefool This is more like a `react-native` problem itself. Try to remove `ios/build` and run build again. This also looks like an old problem which happened in previous `react-native`.

Maybe [this reply from me](https://github.com/dooboolab/react-native-iap/issues/14#issuecomment-360330732) might help too.

**Comment by booboothefool** [comment]
@hyochan I think I was able to get past the error. 👍  Seems like my issues are related to Xcode 11 -> Xcode 12?

I am not why this helped, but I did:

- Workspace Settings > Build System: Legacy Build System (Deprecated), Derived Data: Workspace-relative Location
![image](https://user-images.githubusercontent.com/13257645/98492189-0fc9db00-2205-11eb-9fc1-04564a8152d1.png)

Then I started getting:
```
No architectures to compile for (ONLY_ACTIVE_ARCH=YES, active arch=x86_64, VALID_ARCHS=arm64e armv7s arm64 arm7)
```

So I went to Build Settings > Build Active Architectures Only: Debug: No
![image](https://user-images.githubusercontent.com/13257645/98492259-4b64a500-2205-11eb-97e0-f127d9d0396c.png)

Got past RNIap error and now just getting Expo/Amplitude error. 😅 
```
ld: library not found for -lAmplitude-iOS
```

Should probably just use my holiday vacation to download new React Native + Expo and copy all project files over with updated dependencies, haha.

Thanks!
